src/components/AdvanceAnalyticsV2/data/utils.js

```diff
@@ -8,3 +15,91 @@ const simulateURL = (activeTab, chartType) => {
+`granularity` should be one of the following values: 'day', 'week', 'month', 'quarter', 'year'.
+*/
+export const applyGranularity = (timeSeriesData, dateKey, countKey, granularity) => {
+  if (timeSeriesData.length <= 1) {
+    return timeSeriesData;
+  }
+  const modifiedData = [];
+  let modifiedEntry = timeSeriesData[0];
+  // Apply granularity on the given data.
+  for (let i = 1; i < timeSeriesData.length; i++) {
+    const item = timeSeriesData[i];
+    const dateEntry = item[dateKey];
+    const countEntry = item[countKey];
+    const modifiedDate = modifiedEntry[dateKey];
+
+    // If start-of value for the iteration is same as the start-of value for the modified entry, add the count.
+    if (dayjs.utc(dateEntry).startOf(granularity).isSame(dayjs.utc(modifiedDate).startOf(granularity))) {
+      modifiedEntry[countKey] += countEntry;
+    } else {
+      modifiedData.push(modifiedEntry);
+      modifiedEntry = item;
+    }
+  }
+  modifiedData.push(modifiedEntry);
+  return modifiedData;
+};
+
+/*
+Calculate moving average for the given time series data.
+
+period will denote the number of periods to consider for the moving average calculation.
+e.g. 3 would mean 3-day moving average.
+*/
+const calculateMovingAverage = (timeSeriesData, countKey, period) => {
+  const modifiedData = [];
+  for (let i = 0; i < timeSeriesData.length; i++) {
+    // Set start to 0 for the first `N` entries where `N` is the `period`.
+    const start = i < period ? 0 : (i - period + 1);
+    const end = i + 1;
+
+    // Calculate moving average for the given period.
+    const arraySlice = timeSeriesData.slice(start, end).map((item) => item[countKey]);
+    const movingAverage = sum(arraySlice) / arraySlice.length;
+    modifiedData.push({ ...timeSeriesData[i], [countKey]: movingAverage });
+  }
+  return modifiedData;
+};
+
+/*
+Apply calculation on the given time series data.
+
+calculation should be one of the following values:
+  'total', 'running_total', 'moving_average_3_periods', 'moving_average_7_periods'.
+*/
+export const applyCalculation = (timeSeriesData, countKey, calculation) => {
+  // Noting to do if calculation is TOTAL.
+  if (calculation === CALCULATION.TOTAL) {
+    return timeSeriesData;
+  }
+
+  // Apply running total calculation
+  if (calculation === CALCULATION.RUNNING_TOTAL) {
+    let runningTotal = 0;
+    const modifiedData = [];
+    for (let i = 0; i < timeSeriesData.length; i++) {
+      runningTotal += timeSeriesData[i][countKey];
+      modifiedData.push({ ...timeSeriesData[i], [countKey]: runningTotal });
+    }
+    return modifiedData;
+  }
+
+  // Apply moving average calculation
+  if (calculation === CALCULATION.MOVING_AVERAGE_3_PERIODS) {
+    return calculateMovingAverage(timeSeriesData, countKey, 3);
+  }
+  if (calculation === CALCULATION.MOVING_AVERAGE_7_PERIODS) {
+    return calculateMovingAverage(timeSeriesData, countKey, 7);
+  }
+
+  // If no calculation is applied,
+  return timeSeriesData;
+};
```

Comment:
1. Can we convert the utility method docs to jsdoc?
2. Can we add more details in docs to explain what exactly each util does?